setup.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import os
+from setuptools import setup, find_packages
+from cmake_build_extension import BuildExtension, CMakeExtension
+
+# Read the contents of your README file
+this_directory = os.path.abspath(os.path.dirname(__file__))
+with open(os.path.join(this_directory, "README.md"), encoding='utf-8') as f:
+    long_description = f.read()
+
+setup(
+    name="bipedal_locomotion_framework",
+    author="Diego Ferigo",
+    author_email="diego.ferigo@iit.it",
+    description="Suite of libraries for achieving bipedal locomotion on humanoid robots.",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/dic-iit/bipedal-locomotion-framework",
+    keywords="robot robotics locomotion humanoid framework",
+    license="LGPL",
+    platforms="any",
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
+        "Operating System :: POSIX :: Linux",
+        "Operating System :: MacOS",
+        "Operating System :: Microsoft :: Windows",
+        "Framework :: Robot Framework",
+        "Intended Audience :: Science/Research",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Education",
+        "Programming Language :: C++",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3 :: Only",
+        "License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)",
+    ],
+    use_scm_version=dict(local_scheme="dirty-tag"),
+    setup_requires=["setuptools_scm"],
+    python_requires='>=3.6',
+    install_requires=[
+        "manifpy",
+        "pybind11",
+        "idyntree>=3.0.3.dev0",
+    ],
+    extras_require=dict(
+        test=[
+            "pytest",
+            "pytest-icdiff",
+        ]
+    ),
+    ext_modules=[
+        CMakeExtension(name='InstallAllTargets',
+                       install_prefix="bipedal_locomotion_framework",
+                       cmake_depends_on=["idyntree", "pybind11"],
```

Comment:
It seems fine, even though the vendored osqp included in the casadi wheel has malformed CMake import files (they reference the Travis CI setup). This is not a big deal since in any case for blf we install osqp from sources in the debian container in CI. The only caveat is that we have to specify in `setup.py` the `osqp_DIR` to load the system osqp otherwise by default the casadi osqp is detected by CMake.